mlflow/server/js/src/components/CompareRunScatter.js

```diff
@@ -0,0 +1,222 @@
+import './CompareRunScatter.css';
+
+class CompareRunScatter extends Component {
+  static propTypes = {
+    runInfos: PropTypes.arrayOf(RunInfo).isRequired,
+    metricLists: PropTypes.arrayOf(Array).isRequired,
+    paramLists: PropTypes.arrayOf(Array).isRequired,
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.metricKeys = CompareRunScatter.getKeys(this.props.metricLists);
+    this.paramKeys = CompareRunScatter.getKeys(this.props.paramLists);
+
+    if(this.paramKeys.length + this.metricKeys.length < 2)
+      this.state = {disabled: true};
+    else
+      this.state = {
+        disabled: false,
+        x: this.paramKeys.length > 0 ?
+          {
+            key: this.paramKeys[0],
+            isMetric: false
+          } : {
+            key: this.metricKeys[1],
+            isMetric: true
+          },
+        y: this.metricKeys.length > 0 ?
+          {
+            key: this.metricKeys[0],
+            isMetric: true
+          } : {
+            key: this.paramKeys[1],
+            isMetric: false
+          }
+      }
+  }
+
+  /**
+   * Find in a list of metrics/params a certain key
+   */
+  static findInList(data, key) {
+    let found = undefined;
+    data.forEach((value) => {
+      if(value.key == key)
+        found = value;
+    });
+    return found;
+  }
+
+  static getKeys(lists) {
+    let keys = {};
+    lists.forEach((list) => 
+      list.forEach((item) => {
+        if(!(item.key in keys))
+          keys[item.key] = true;
+        if(isNaN(parseFloat(item.value)))
+          keys[item.key] = false;
+      }
+    ));
+    return Object.keys(keys).filter(k => keys[k]).sort();
+  }
+
+  // Get the value of the metric/param described by {key, isMetric}, in run i
+  getValue(i, {key, isMetric}) {
+    const value = CompareRunScatter.findInList(
+      (isMetric?this.props.metricLists:this.props.paramLists)[i],
+      key);
+    return value === undefined ? value : value.value;
+  }
+
+  render() {
+    if(this.state.disabled)
+      return <div></div>
+
+    const scatterData = [];
+
+    this.props.runInfos.forEach((_, index) => {
+      const x = this.getValue(index, this.state.x);
+      const y = this.getValue(index, this.state.y);
+      if(x === undefined || y === undefined)
+        return;
+      scatterData.push({index, x: +x, y: +y});
+    });
+
+    return (<div>
+      <h2>Scatter Plot</h2>
+      <div className="container">
+        <div className="row">
+          <form className="col-xs-3">
+            <div className="form-group">
+              <label htmlFor="y-axis-selector">X-axis:</label>
+              {this.renderSelect("x")}
+            </div>
+            <div className="form-group">
+              <label htmlFor="y-axis-selector">Y-axis:</label>
+              {this.renderSelect("y")}
+            </div>
+          </form>
+          <div className="col-xs-9">
+            <ResponsiveContainer width="100%" aspect={1.55}>
```

Comment:
For some reason, this doesn't resize properly at some window sizes, such as width of 800px; instead it seems to jump in discrete increments, maybe because of the reactive grid stuff around it.